### ISSUE TITLE
Multi-user case setup issues

### ISSUE THREAD

**Issue by Psylence7** [sleuthkit/autopsy#3255]
I've seen a previous issue opened for this with no response so I'm hoping I might have more luck. I have configured Solr, Postgres and ActiveMQ on my server, entered the credentials into the Multi-user tab in Options and they've all been accepted. 

My issue now is that I can't generate a case all I get is the error "Failed to get lock on case directory". The machines are on the same network, can connect to each other, sharing permissions are setup and the folder isn't Read-Only, and I made sure to use UNC paths when choosing the shared folder for the case on the off-chance that was an issue. Autopsy will go as far as creating a subfolder for the case if it doesn't exist, but after 1 minute the process fails and gives me that error.
![image](https://user-images.githubusercontent.com/5718240/33563565-3e4a4654-d910-11e7-9289-193b1d502c4c.png)

Any thoughts would be appreciated as I'm completely stumped why Autopsy can't seem to lock onto the folder properly.

**Comment by rcordovano** [comment]
This error message means that the coordination service used for multi-user cases either timed out getting a lock for the case, experienced an error, or is not running.

The coordination service is implemented in terms of the ZooKeeper instance that is run by the Solr server. Is it safe to assume you installed the Apache Solr 4.10.3-0 installation package from Bitnami as described in the documentation? 

One way to check that the coordination service is up and running is to open up a Solr Admin page for the Solr server. Select "Cloud" and "Tree" and you should see something like this:

![image](https://user-images.githubusercontent.com/1713605/33572527-35beb5ec-d901-11e7-931b-710c8a36ada5.png)

The /autopsy/cases part of the tree is where you would see the ZooKeeper nodes that are used to implement the locks  on cases.

Another thing to look at is the "between cases" Autopsy log with stack traces.  The log is split across files with names like autopsy_traces.log.[number]. There should be error messages in the log of the form "Error opening [CASE DISPLAY NAME]  ([CASE UNIQUE NAME]) in [CASE DIRECTORY] as the current case" with a stack trace. If you are able to post an instance of this log message, I will take a look at it for you.

**Comment by Psylence7** [comment]
@rcordovano 
Thanks in advance for this, I will have a look tomorrow when I'm back on-site. You are correct, I followed the documentation to the letter, sourcing older versions of the tools required. I initially tried with the newest versions of each but even getting Solr to function as required I found difficult, so a rollback was the best option. I will let you know how I get on.

**Comment by Psylence7** [comment]
@rcordovano 
I've accessed the Solr installation and I appear to have a different layout to yourself?
![image](https://user-images.githubusercontent.com/5718240/33602947-ce779cc0-d9a8-11e7-93ca-157ef6040559.png)
I'm unsure of whether I've missed something during the installation that's caused this but I appear to be using the recommended version as far as I can tell. I tried to jump to the write menu by appending "~cloud" into the address bar and got the error:

**Loading of "/solr/zookeeper?wt=json" failed (HTTP-Status 404)
"Zookeeper is not configured for this Solr Core. Please try connecting to an alternate zookeeper address."**

I would have said I'd followed the documentation and installer to the letter, is it worth pulling out Solr and starting again?

**Comment by Psylence7** [comment]
Following on from this I've reviewed the documentation from 4.10 and found out more information on the SolrCloud functionality that I appear to be missing, and when trying to access the Cloud port on 9983 (8983+1000 as per documentation) it refuses the connection. At this point I feel a reinstall may be the best course of action but I will wait and see if you can offer any better insight into why this might have occurred or how I can rectify it.

**Comment by rcordovano** [comment]
@Psylence7, I am going to turn this over to @esaunders for further discussion, since he has much more experience than me when it comes to Solr set up.

**Comment by esaunders** [comment]
Yes, it appears that ZooKeeper is not running on your Solr server (the absence of "Cloud" on the left side of the screenshot you sent is one clue). Are you sure you have added ++JvmOptions=-DzkRun to the serviceinstall.bat script? Feel free to send along the contents of your serviceinstall.bat script. Note that if you wish to make changes to this script after you have installed the service you will need to uninstall and reinstall the service.

**Comment by Psylence7** [comment]
@esaunders 
Thanks for your reply, it appears I have not done that, and I have just realised that the documentation I have been working off is the 4.1 guide which does not include that step. Having a quick look it appears the only place I found a link to the documentation was from the Wiki, which only links to the 4.1 documentation. I shall compare both guides and likely have to start from scratch in that case. Many thanks at the very least for helping me spot that!

**Comment by esaunders** [comment]
FYI: I followed the documentation link from http://www.sleuthkit.org/autopsy/docs.php  which took me to http://sleuthkit.org/autopsy/docs/user-docs/4.3/

**Comment by Psylence7** [comment]
@esaunders 
I edited the address bar (as the website doesn't appear to have "options -Indexes" disabled) and you can see the full version list:
http://sleuthkit.org/autopsy/docs/user-docs/

So actually the latest documentation should be:
https://sleuthkit.org/autopsy/docs/user-docs/4.5.0/

**Comment by esaunders** [comment]
Yes, 4.5 is the latest version. I've notified @bcarrier and hopefully the outdated links on various sites will be taken care of soon.

**Comment by Psylence7** [comment]
@esaunders 
Well I've followed the new instructions to the letter and I'm still having a couple of issues I'm afraid, with the same error. My Solr dashboard now looks correct, however I don't have a Config or Manifests folder, only files. Does this suggest an error with Zookeeper
![image](https://user-images.githubusercontent.com/5718240/33663628-a68ecc66-da88-11e7-85c3-42c0f0455093.png)
Following this, I've also dug out one of the tracelog files and attached it as **rcordovano** suggested to be reviewed (copied into a TXT file). I can see the errors in there indicating the failure to both open and create a case, but the details are beyond me I'm afraid.
[tracelog.txt](https://github.com/sleuthkit/autopsy/files/1535287/tracelog.txt)
As before, any assistance would be appreciated.

**Comment by esaunders** [comment]
Can you send along a screenshot of what appears when you select the "Logging" node in the Solr Admin interface?
One thing that stands out in the screenshot above is that the path to your case output folder appears to be a local drive (i.e. C:). For multi user cases the case output folder path must be a network share that is accessible to the user account under which the Solr service is running.

**Comment by Psylence7** [comment]
@esaunders 
Yes, I'm aware of the C: directory anomaly. I'm actually using a slightly unorthodox setup with Autopsy and Solr/Postgres/ActiveMQ on the same machine, my plan being to run an initial Autopsy case creation on this machine with the multi-user tools running in the background, and allow other end-users to connect to that main machine which I'll be using for analysis as it's a server. Can you see any reason why this may not work?
![image](https://user-images.githubusercontent.com/5718240/33669946-b35abc44-da9b-11e7-9443-8f17a388cb2e.png)

**Comment by esaunders** [comment]
I can't think of any reason off the top of my head why that setup wouldn't work.
What has me confused however is that the log file you sent earlier indicates that Autopsy cannot access ZooKeeper but the screenshot you sent earlier clearly shows ZooKeeper elements under the "cases" node in the tree.
Can you send along the contents of the C:\Bitnami\solr-4.10.3-0\apache-solr\logs folder and any solr.log.* files that exist on the system? Feel free to zip them up and email them to me at esaunders at basistech.com if you would prefer not to post them here.

**Comment by Psylence7** [comment]
@esaunders 
Thanks for that, I've emailed the log files in a .7z from my Gmail account which you should receive shortly.

**Comment by esaunders** [comment]
I'm not seeing anything obvious in those logs either. Perhaps the best course of action at this point is to get back to a clean slate by (a) stopping the Solr service (b) deleting the ZooKeeper database, i.e. delete the C:\Bitnami\solr-4.10.3-0\apache-solr\solr\zoo_data\version-2 folder (c) delete existing case output data from C:\DATA\MULTOPSY (d) re-starting the Solr service and finally (e) creating a new case using a case name not used before.
Let me know how it goes.

**Comment by Psylence7** [comment]
@esaunders 
I'l pick this up again in the morning when I'm back on site and let you know.

**Comment by Psylence7** [comment]
@esaunders 
I followed all steps as you advised and I'm afraid it's still not working. Same error as before, I'm not sure what could be causing it to time out at this stage.

**Comment by esaunders** [comment]
Without having access to the machine the logs are our best hope for
figuring this out. If you send me all log files from
%APPDATA%/autopsy/var/logs and the case logs folder along with the latest
Solr logs I will take a look.

**Comment by Psylence7** [comment]
@esaunders 
Many thanks, I've emailed the logs over as before and hopefully await some good news from you that I've just messed something up, simple or otherwise, that would explain this.

**Comment by esaunders** [comment]
Apologies for the delay in getting back to you but the only thing that stands out as abnormal in the logs you sent is the fact that Autopsy appears to lose connectivity to ZooKeeper for some unknown reason.
I'm reaching here but is it possible that there is some piece of protective software on your system (e.g. anti-malware, anti-virus) that has the ability to forcibly close connections?
